portality/forms/application_processors.py

```diff
@@ -303,6 +303,10 @@ def finalise(self, account, save_target=True, email_alert=True):
             raise Exception("You cannot edit a not-existent application")
         if self.source.application_status == constants.APPLICATION_STATUS_ACCEPTED:
             raise Exception("You cannot edit applications which have been accepted into DOAJ.")
+        if self.form.application_status.data == "accepted" and (
+                self.source.current_journal is None or not models.Journal.pull(
+            self.source.current_journal).is_in_doaj()):
```

Comment:
This line could fail if `Journal.pull` returns None, I would be inclined to break this out into a block rather than inlining it into the if statement

```python
if self.form.application_status.data == "accepted" and self.source.current_journal is not None:
    j = models.journal.pull(self.source.current_journal)
    if j is None or not j.is_in_doaj():
        raise Exception()
```

Reply:
Generally speaking I recommend simple if statements, and to prep all the things you need beforehand.  Inlining a load from database is a good warning sign that the if statement is too complex